SourceCode/include/tracer/RayTracer.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include <string>
+#include <vector>
+
+#include "Image.h"
+#include "Ray.h"
+#include "SceneParser.h"
+#include "Triangle.h"
+#include "tracer/Camera.h"
+
+class RayTracer {
+ private:
+  bool rayUpdateRequired;
+  Image image;
+  Camera camera;
+  SceneParser sceneParser;
```

Comment:
The parsing of the scene should be done separately from the Ray Tracing rendering logic

Reply:
I am not sure if I understand you properly. Do you mean that the `SceneParser` instance shouldn't be a member of the `RayTracer` class and I should supply the collection of triangles when I generate the image, instead?